pages/docs/tooling/tooling-builtin.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Built-in tooling
+permalink: tooling-builtin.html
+keywords: tooling, xml, configuration, version
+summary: "Built-in tooling is always installed alongside preCICE and provides some basic functionality."
+---
+
+Part of a preCICE installation is the tool `binprecice`.
+It provides an easy-to-use interface to tooling API of the preCICE library.
+
+## XML reference
+
+```bash
+binprecice md
+binprecice xml
+binprecice dtd
+```
+
+This prints the XML reference to the console in various flavours.
+
+### `md`
+
+This prints the XML configuration in Markdown format.
+You can find the reference of the latest release [on the website](configuration-xml-reference.html).
+
+It is possible to generate a local version of the reference by rendering the Markdown to HTML using the `markdown` command.
+Be aware that this version does not contain styling, LaTeX rendering, and functioning links.
+
+```bash
+binprecice md | markdown > reference.html
+```
+
+### `xml`
+
+This prints the XML configuration with in-lined annotations of tags and attributes.
+
+### `dtd`
+
+This prints the DTD information, which can be used to validate the XML configuration file.
+
+## preCICE version
+
+{% version 2.4.0 %}
+This feature is available since version 2.4.0.
+{% endversion %}
+
+```bash
+binprecice version
+```
+
+This prints the version information of preCICE, which consists of multiple semicolon-separated parts.
+
+1. the version of preCICE e.g. `2.3.0`.
+2. the git revision of this version e.g. `v2.3.0-87-g04ee7308-dirty`.
+   This is interesting for development versions of preCICE.
+   It is not available if the library wasn't build using the git repository.
+3. Configuration options for MPI, PETSc, Python and some more.
+4. Compilation and linker flags used to build preCICE
+
+## Configuration check
+
+{% version 2.4.0 %}
+This feature is available since version 2.4.0.
+{% endversion %}
+
+```bash
+binprecice check FILE [ PARTICIPANT [ COMMSIZE ] ]
+```
+
+The `check` runs the preCICE configuration parsing and checking logic on the given configuration file.
```

Comment:
We have to be careful to not oversell the feature here. A user might think that the call also checks whether the exchange tags are set correctly or similar. It "only" validates whether the configuration follows the semantics, not the "logic" of  the configuration.

Reply:
Maybe also give an example of issues it can detect and an example of issues it cannot detect?

Reply:
True, some more information would be appropriate here.

> A user might think that the call also checks whether the exchange tags are set correctly or similar. It "only" validates whether the configuration follows the semantics, not the "logic" of the configuration.

This is a bit wishy washy at the moment and mostly due to how our configuration works.

As an example the config parser checks if the participants of the exchange tags are part of the cplscheme and whether they hold the data, but doesn't ensure that all required data is transferred.
It checks "most of the local logic", if that makes sense.

Reply:
I added some clarifications